packages/firestore/src/api/credentials.ts

```diff
@@ -175,11 +188,25 @@ export class FirebaseCredentialsProvider implements CredentialsProvider {
    * The auth token listener registered with FirebaseApp, retained here so we
    * can unregister it.
    */
-  private tokenListener: ((token: string | null) => void) | null = null;
+  private tokenListener: (() => void) | null = null;
 
   /** Tracks the current User. */
   private currentUser: User = User.UNAUTHENTICATED;
-  private receivedInitialUser: boolean = false;
+
+  /**
+   * Promise that allows blocking on the next `tokenChange` event. The Promise
+   * is re-assgined in `awaitTokenAndRaiseInitialEvent()` to allow blocking on
+   * an a lazily loaded Auth instance. In this case, `this.receivedUser`
+   * resolves once when the SDK first detects that there is no synchronous
+   * Auth, and then gets re-created and resolves again once Auth is loaded.
+   */
+  private receivedUser = new Deferred();
+
+  /**
+   * Whether the initial token event has been raised. This can go back to
+   * `false` if Firestore first starts without Auth and Auth is loaded later.
+   */
+  private initialEventRaised: boolean = false;
```

Comment:
This variable doesn't seem to do anything, because it's only set in `awaitTokenAndRaiseInitialEvent` which is synchronous. It's not used in `awaitTokenAndRaiseInitialEvent()` itself and is always set to true at the end of the function, so it seems the outside world will always see this variable as `true`.